internal/services/nginx/nginx_certificate_data_source.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package nginx
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/nginx/2023-04-01/nginxcertificate"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/nginx/2023-04-01/nginxdeployment"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type CertificateDataSourceModel struct {
+	Name                   string `tfschema:"name"`
+	NginxDeploymentId      string `tfschema:"nginx_deployment_id"`
+	KeyVirtualPath         string `tfschema:"key_virtual_path"`
+	CertificateVirtualPath string `tfschema:"certificate_virtual_path"`
+	KeyVaultSecretId       string `tfschema:"key_vault_secret_id"`
+}
+
+type CertificateDataSource struct{}
+
+var _ sdk.DataSource = CertificateDataSource{}
+
+func (m CertificateDataSource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"nginx_deployment_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: nginxdeployment.ValidateNginxDeploymentID,
+		},
+	}
+}
+
+func (m CertificateDataSource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"key_virtual_path": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"certificate_virtual_path": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"key_vault_secret_id": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+	}
+}
+
+func (m CertificateDataSource) ModelObject() interface{} {
+	return &CertificateDataSourceModel{}
+}
+
+func (m CertificateDataSource) ResourceType() string {
+	return "azurerm_nginx_certificate"
+}
+
+func (m CertificateDataSource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Nginx.NginxCertificate
+			var model CertificateDataSourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return err
+			}
+			deploymentId, err := nginxdeployment.ParseNginxDeploymentID(model.NginxDeploymentId)
+			if err != nil {
+				return fmt.Errorf("error parsing NGINX deployment ID %s: %+v", deploymentId, err)
```

Comment:
The error returned by the parsing function already contains the information we need. We don't need to wrap it further.
```suggestion
				return err
```

Reply:
👍 made the change.